doc/release/upcoming_changes/24922.new_feature.rst

```diff
@@ -0,0 +1,5 @@
+`numpy.long` and `numpy.ulong`
+------------------------------
+
+`numpy.long` and `numpy.ulong` have been reintroduced as a direct
+mapping to C's ``long`` and ``unsigned long`` types.
```

Comment:
"reintroduced" suggests they have the same meaning as before they were removed, but this is not the case.

Reply:
Maybe we can just remove it and say: ... added as NumPy integers mapping to C's `long` and `unsigned long`.  Prior to NumPy 1.24 `np.long` was an alias to Python's `int`.

It was actually deprecated at the time already I think.  (Ignoring that it was Python `long` on Python 2, and then actually had the `long long` meaning in `dtype(...)`, I think it was just `int` after that.)

Reply:
Sure, I updated it accordingly.